src/metabase/api/setup.clj

```diff
@@ -86,4 +88,108 @@
       (catch Throwable e
         (response-invalid :general (.getMessage e))))))
 
+
+;;; Admin Checklist
+
+(defn- admin-checklist-values []
+  (let [has-dbs?           (db/exists? Database)
+        has-dashboards?    (db/exists? 'Dashboard)
+        has-pulses?        (db/exists? 'Pulse)
+        has-labels?        (db/exists? 'Label)
+        has-hidden-tables? (db/exists? 'Table, :visibility_type [:not= nil])
+        has-metrics?       (db/exists? 'Metric)
+        has-segments?      (db/exists? 'Segment)
+        num-tables         (db/select-one-count 'Table)
+        num-cards          (db/select-one-count 'Card)
+        num-users          (db/select-one-count 'User)]
+    [{:title       "Add a database"
+      :group       "Get connected"
+      :description "TODO - Write something good here"
+      :link        "/admin/databases/create"
+      :completed   has-dbs?
+      :triggered   :always}
```

Comment:
There is nothing specifically wrong with this approach, but it would be my vote that UI specific definitions like this should be hosted and maintained in the frontend app.  The frontend app represents a more natural place for user copy, visual/ux choices, and links to be managed.